platforms/unix/plugins/SqueakSSL/openssl_overlay.h

```diff
@@ -0,0 +1,301 @@
+long (*sqo_SSL_ctrl)(SSL *ssl, int cmd, long larg, void *parg);
+int (*sqo_SSL_get_error)(const SSL *s, int ret_code);
+X509 *(*sqo_SSL_get_peer_certificate)(const SSL *s);
+long (*sqo_SSL_get_verify_result)(const SSL *ssl);
+int (*sqo_SSL_library_init)(void);
+void (*sqo_SSL_load_error_strings)(void);
+SSL *(*sqo_SSL_new)(SSL_CTX *ctx);
+int (*sqo_SSL_read)(SSL *ssl, void *buf, int num);
+void (*sqo_SSL_set_accept_state)(SSL *s);
+void (*sqo_SSL_set_bio)(SSL *s, BIO *rbio, BIO *wbio);
+void (*sqo_SSL_set_connect_state)(SSL *s);
+int (*sqo_SSL_write)(SSL *ssl, const void *buf, int num);
+int (*sqo_X509_NAME_get_text_by_NID)(X509_NAME *name, int nid, char *buf, int len);
+X509_NAME *(*sqo_X509_get_subject_name)(X509 *a);
+void *(*sqo_X509_get_ext_d2i)(X509 *x, int nid, int *crit, int *idx);
+void (*sqo_X509_free)(X509 *ssl);
+
+// OPENSSL_VERSION_NUMBER >= 0x10002000L
+int (*sqo_X509_check_ip_asc)(X509 *x, const char *ipasc, unsigned int flags);
+int (*sqo_X509_check_host)(X509 *x, const char *chk, size_t chklen, unsigned int flags, char **peername);
+
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+_STACK *(*sqo_sk_new_null)(void);
+int (*sqo_sk_push)(_STACK *st, void *data);
+void (*sqo_sk_free)(_STACK *st);
+void *(*sqo_sk_value)(const _STACK *st, int i);
+int (*sqo_sk_num)(const _STACK *st);
+void (*sqo_sk_pop_free)(_STACK *st, void (*func) (void *));
+#else
+STACK *(*sqo_sk_new_null)(void);
+int (*sqo_sk_push)(STACK *st, char *data);
+void (*sqo_sk_free)(STACK *st);
+char *(*sqo_sk_value)(STACK *st, int i);
+int (*sqo_sk_num)(STACK *st);
+void (*sqo_sk_pop_free)(STACK *st, void (*func) (void *));
+#endif // OPENSSL_VERSION_NUMBER >= 0x10000000L
+
+#define sqo_BIO_set_close(b,c)                          \
+  (int)sqo_BIO_ctrl(b,BIO_CTRL_SET_CLOSE,(c),NULL)
+
+#define sqo_SSL_set_tlsext_host_name(s,name)                            \
+  sqo_SSL_ctrl(s,SSL_CTRL_SET_TLSEXT_HOSTNAME,TLSEXT_NAMETYPE_host_name,(char *)name)
+
+#define sqo_SSL_CTX_set_options(ctx,op)                 \
+  sqo_SSL_CTX_ctrl((ctx),SSL_CTRL_OPTIONS,(op),NULL)
+
+#define sqo_SKM_sk_num(type, st)                \
+  sqo_sk_num(CHECKED_STACK_OF(type, st))
+#define sqo_SKM_sk_value(type, st,i)                    \
+  ((type *)sqo_sk_value(CHECKED_STACK_OF(type, st), i))
+#define sqo_SKM_sk_free(type, st)               \
+  sqo_sk_free(CHECKED_STACK_OF(type, st))
+#define sqo_SKM_sk_pop_free(type, st, free_func)                        \
+  sqo_sk_pop_free(CHECKED_STACK_OF(type, st), CHECKED_SK_FREE_FUNC(type, free_func))
+#define sqo_sk_GENERAL_NAME_num(st)             \
+  sqo_SKM_sk_num(GENERAL_NAME, (st))
+#define sqo_sk_GENERAL_NAME_value(st, i)        \
+  sqo_SKM_sk_value(GENERAL_NAME, (st), (i))
+#define sqo_sk_GENERAL_NAME_free(st)            \
+  sqo_SKM_sk_free(GENERAL_NAME, (st))
+#define sqo_sk_GENERAL_NAME_pop_free(st, free_func)     \
+  sqo_SKM_sk_pop_free(GENERAL_NAME, (st), (free_func))
+
+#if !defined(X509_CHECK_FLAG_SINGLE_LABEL_SUBDOMAINS)
```

Comment:
Could you add a comment when this is not defined? Otherwise, prefix it as well?
-> `sqo_X509_CHECK_FLAG_SINGLE_LABEL_SUBDOMAINS`